### PR TITLE
[8.x] Add a default timeout in the HTTP client

### PR DESCRIPTION
**Pull request by themsaid**
## Update

Moved to the 9.x release: https://github.com/laravel/framework/pull/40187

### CODE REVIEW

File: src/Illuminate/Http/Client/PendingRequest.php [hunk 1]
```diff
@@ -162,6 +162,7 @@ public function __construct(Factory $factory = null)
 
         $this->options = [
             'http_errors' => false,
+            'timeout' => 10,
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Client/PendingRequest.php:165]: 10 seconds is probably too low for a lot of people. I'd recommend going with:

```
'connect_timeout' => 10,
'timeout' => 15,
```

like in https://github.com/GrahamCampbell/Guzzle-Factory/blob/5.0/src/GuzzleFactory.php.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by mabdullahsari** [comment]
Please target 9.x. 

This change will break an app I work on because it has API calls that last 3 mins (I did not design it 🙃).

**Comment by SjorsO** [comment]
A default timeout is a great idea. I've had queue workers get stuck because HTTP calls timed-out in a weird way (even though supervisor reported they were working fine).

**Comment by leonardovee** [comment]
A default timeout is sure a good idea, but shouldn't this be treat as a breaking change?

**Comment by taylorotwell** [comment]
@themsaid I agree we should probably send this to 9.x / master.

**Pull request closed without merging**